### PR TITLE
Ble rssi svc

### PR DESCRIPTION
**Pull request by Lumpusz**
## Description:

**Related issue (if applicable):** fixes <link to issue>
https://github.com/esphome/feature-requests/issues/94
**Pull request in [esphome](https://github.com/esphome/esphome) with YAML changes (if applicable):** esphome/esphome#<esphome PR number goes here>
https://github.com/esphome/esphome/pull/800
## Checklist:

  - [ x] Branch: `next` is for changes and new documentation that will go public with the next ESPHome release. Fixes, changes and adjustments for the current release should be created against `current`.
  - [x ] Link added in `/index.rst` when creating new documents for new components or cookbook.

### CODE REVIEW

File: components/binary_sensor/ble_presence.rst [hunk 1]
```diff
@@ -22,13 +22,25 @@ bluetooth low energy device.
...
+        name: "ESP32 BLE Tracker Test Service 128 bit"
 
 Configuration variables:
 ------------------------
 
--  **mac_address** (**Required**, MAC Address): The MAC address to track for this
-   binary sensor.
 -  **name** (**Required**, string): The name of the binary sensor.
+-  **mac_address** (*Optional*, MAC Address): The MAC address to track for this
+   binary sensor. Either this or **service_uuid** has to be present.
```
Comment (OttoWinter) [inline components/binary_sensor/ble_presence.rst]:
```suggestion
   binary sensor. Either this or ``service_uuid`` has to be present.
```

File: components/binary_sensor/ble_presence.rst [hunk 2]
```diff
@@ -22,13 +22,25 @@ bluetooth low energy device.
...
+-  **service_uuid** (*Optional*, 16 bit, 32 bit, or 128 bit BLE Service UUID): The BLE
+   Service UUID which can be tracked if the device randomizes the MAC address. Either
+   this or **mac_address** has to be present.
```
Comment (OttoWinter) [inline components/binary_sensor/ble_presence.rst]:
```suggestion
   this or ``mac_address`` has to be present.
```

File: components/binary_sensor/ble_presence.rst [hunk 3]
```diff
@@ -22,13 +22,25 @@ bluetooth low energy device.
...
 
 Configuration variables:
 ------------------------
 
--  **mac_address** (**Required**, MAC Address): The MAC address to track for this
-   binary sensor.
 -  **name** (**Required**, string): The name of the binary sensor.
+-  **mac_address** (*Optional*, MAC Address): The MAC address to track for this
+   binary sensor. Either this or **service_uuid** has to be present.
+-  **service_uuid** (*Optional*, 16 bit, 32 bit, or 128 bit BLE Service UUID): The BLE
```
Comment (OttoWinter) [inline components/binary_sensor/ble_presence.rst]:
```suggestion
-  **service_uuid** (*Optional*, string): The BLE
```

Usually, the second parameter here shows what YAML type it is (though that seems to not have been enforced with MAC address, feel free to change that to `string` too).

File: components/binary_sensor/ble_presence.rst [hunk 4]
```diff
@@ -22,13 +22,25 @@ bluetooth low energy device.
       - platform: ble_presence
         mac_address: AC:37:43:77:5F:4C
         name: "ESP32 BLE Tracker Google Home Mini"
+      - platform: ble_presence
```
Comment (OttoWinter) [inline components/binary_sensor/ble_presence.rst:27]: 1. Please only add one example for service_uuid here. The example at the top should always be minimal to give the user an idea. You can create a `.. note::`` below with more examples with different uuid lengths.

2. Please put a comment above each of these two entries (mac and uuid). One saying `# Presence based on MAC address` and the other the same but with UUID.

File: guides/supporters.rst [hunk 5]
```diff
@@ -52,6 +52,7 @@ You guys are awesome!
 - DrZzs
 - fabi
 - Franck Nijhof
+- Gaiar Baimuratov
```
Comment (OttoWinter) [inline guides/supporters.rst]: It looks like you created this PR from the `current` branch. Please rebase this PR on the `next` branch (see contributing docs)

File: components/binary_sensor/ble_presence.rst [hunk 6]
```diff
@@ -55,7 +70,11 @@ address type and advertised name. If you don't see these messages, your device i
 currently not supported.
 
 Please note that devices that show a ``RANDOM`` address type in the logs cannot be used for
-tracking, since their MAC-address periodically changes.
+MAC address based tracking, since their MAC-address periodically changes. Instead you can create
+a BLE beacon, set a unique 16 bit, 32 bit or 128 bit Service UUID and track your device based on that.
+Make sure you don't pick a [GATT Service UUID](https://www.bluetooth.com/specifications/gatt/services/),
```
Comment (OttoWinter) [inline components/binary_sensor/ble_presence.rst]: That format is markdown - this page is written in RST - see contributing guide for creating links with RST.

File: components/binary_sensor/ble_presence.rst [hunk 7]
```diff
@@ -19,16 +19,31 @@ bluetooth low energy device.
...
         name: "ESP32 BLE Tracker Google Home Mini"
+      # Presence based on BLE Service UUID
+      - platform: ble_presence
+        service_uuid: '11aa'
+        name: "ESP32 BLE Tracker Test Service 16 bit"
+
+.. note::
+
+    Service UUID can be 16 bit long, as in the example, but it can also be 32 bit long
+    like `1122aaff`, or 128 bit long like `11223344-5566-7788-99aa-bbccddeeff00`.
```
Comment (OttoWinter) [inline components/binary_sensor/ble_presence.rst]:
```suggestion
    like ``1122aaff``, or 128 bit long like ``11223344-5566-7788-99aa-bbccddeeff00``.
```

Same here, this is RST syntax.

File: components/binary_sensor/ble_presence.rst [hunk 8]
```diff
@@ -19,16 +19,31 @@ bluetooth low energy device.
...
+
 
 Configuration variables:
 ------------------------
 
--  **mac_address** (**Required**, MAC Address): The MAC address to track for this
-   binary sensor.
 -  **name** (**Required**, string): The name of the binary sensor.
+-  **mac_address** (*Optional*, MAC Address): The MAC address to track for this
+   binary sensor. Either this or ''service_uuid'' has to be present.
```
Comment (OttoWinter) [inline components/binary_sensor/ble_presence.rst]:
```suggestion
   binary sensor. Either this or ``service_uuid`` has to be present.
```

File: components/binary_sensor/ble_presence.rst [hunk 9]
```diff
@@ -19,16 +19,31 @@ bluetooth low energy device.
...
+-  **service_uuid** (*Optional*, string) 16 bit, 32 bit, or 128 bit BLE Service UUID
+   which can be tracked if the device randomizes the MAC address. Either
+   this or ''mac_address'' has to be present.
```
Comment (OttoWinter) [inline components/binary_sensor/ble_presence.rst]:
```suggestion
   this or ``mac_address`` has to be present.
```

File: components/binary_sensor/ble_presence.rst [hunk 10]
```diff
@@ -55,7 +70,11 @@ address type and advertised name. If you don't see these messages, your device i
 currently not supported.
 
 Please note that devices that show a ``RANDOM`` address type in the logs cannot be used for
-tracking, since their MAC-address periodically changes.
+MAC address based tracking, since their MAC-address periodically changes. Instead you can create
+a BLE beacon, set a unique 16 bit, 32 bit or 128 bit Service UUID and track your device based on that.
+Make sure you don't pick a .. _GATT Service UUID: https://www.bluetooth.com/specifications/gatt/services/,
```
Comment (OttoWinter) [inline components/binary_sensor/ble_presence.rst]:
```suggestion
Make sure you don't pick a `GATT Service UUID <https://www.bluetooth.com/specifications/gatt/services/>`__,
```

File: guides/supporters.rst (hunk 5 above)
Comment (OttoWinter) [reply guides/supporters.rst]: This still applies

**Comment by Lumpusz** [comment]
I did rebase my branch based on 'next' and force pushed the changes.

**Pull request merged**